shared/eip712/eip712.go

```diff
@@ -0,0 +1,54 @@
+package eip712
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type EIP712Components struct {
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+	V uint8    `json:"v"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e EIP712Components) String() {
```

Comment:
rename this to Print(). Functions named String() should return a string to fulfill <https://pkg.go.dev/fmt#Stringer>

however, i think we want the Stringer interface to be implemented as:
```golang
func (e EIP712Components) String() string {
    out, err := e.MarshalText()
    if err != nil {
        // MarshalText should never return an error
        panic(err)
    }
    return string(out)
}
```
so that the "native" format is the 65 byte 0x-prefixed hex string.